source/grid/grid_generator.cc

```diff
@@ -28,11 +29,1219 @@
+         * points in total (leading and trailing edge points are same)
+         *
+         *             x            x
+         *     x                                  x
+         *  xo                                            xo
+         *    o                              o
+         *         o          o
+         *
+         * @param[in] centerpoint Indicates joukowski circle center coordinate.
+         * @param[in] number_points Required equidistant airfoil points.
+         * @param[in] factor Stating the relation
+         *  provided_non_equidistant points/required_equidistant_points.
+         * @return airfoil_1D Array of vectors with upper and lower
+         *  airfoil points.
+         */
+        static std::array<std::vector<Point<2>>, 2>
+        joukowski(const Point<2>     centerpoint,
+                  const unsigned int number_points,
+                  const unsigned int factor)
+        {
+          std::array<std::vector<Point<2>>, 2> airfoil_1D;
+          const unsigned int total_points    = 2 * number_points - 2;
+          const unsigned int n_airfoilpoints = factor * total_points;
+          // joukowski points on the entire airfoil, i.e. upper and lower side
+          const auto jouk_points =
+            joukowski_transform(joukowski_circle(centerpoint, n_airfoilpoints));
+
+          // vectors to collect airfoil points on either upper or lower side
+          std::vector<Point<2>> upper_points;
+          std::vector<Point<2>> lower_points;
+
+          {
+            // find point on nose and point on tail
+            unsigned int nose_index        = 0;
+            unsigned int tail_index        = 0;
+            double       nose_x_coordinate = 0;
+            double       tail_x_coordinate = 0;
+
+
+            // find index in vector to nose point (min) and tail point (max)
+            for (unsigned int i = 0; i < jouk_points.size(); i++)
+              {
+                if (jouk_points[i](0) < nose_x_coordinate)
+                  {
+                    nose_x_coordinate = jouk_points[i](0);
+                    nose_index        = i;
+                  }
+                if (jouk_points[i](0) > tail_x_coordinate)
+                  {
+                    tail_x_coordinate = jouk_points[i](0);
+                    tail_index        = i;
+                  }
+              }
+
+            // copy point on upper side of airfoil
+            for (unsigned int i = tail_index; i < jouk_points.size(); i++)
+              upper_points.emplace_back(jouk_points[i]);
+            for (unsigned int i = 0; i <= nose_index; i++)
+              upper_points.emplace_back(jouk_points[i]);
+            std::reverse(upper_points.begin(), upper_points.end());
+
+            // copy point on lower side of airfoil
+            for (unsigned int i = nose_index; i <= tail_index; i++)
+              lower_points.emplace_back(jouk_points[i]);
```

Comment:
You can probably do this more efficiently by saying
```
  lower_points.insert (lower_points.end(), jouk_points.begin()+nose_index, jouk_points.begin()+tail_index+1);
```

Reply:
Cool! But I don't understand why for input last, i.e. `jouk_points.begin()+tail_index+1`, the +1 is required. The last input element should be `jouk_points[tail_index]`.

Reply:
The `insert` functions, like many other C++ library functions and conventions in deal.II, work with half-open intervals, i.e., ranges where the `begin` index is included but the `end` index is excluded.

Reply:
Thanks!